test/rekt/resources/sinkbinding/sinkbinding_test.go

```diff
@@ -141,3 +144,44 @@ func Example_full() {
 	//         match1: this
 	//         match2: that
 }
+
+func Example_withSink() {
+	ctx := testlog.NewContext()
+	images := map[string]string{}
+	cfg := map[string]interface{}{
+		"name":      "foo",
+		"namespace": "bar",
+	}
+
+	sinkRef := &duckv1.Destination{
+		Ref: &duckv1.KReference{
+			Kind:       "sinkkind",
+			Namespace:  "sinknamespace",
+			Name:       "sinkname",
+			APIVersion: "sinkversion",
+		},
+		URI: &apis.URL{Path: "uri/parts"},
+         }
+         sinkbinding.WithSink(sinkRef)(cfg)
+	    files, err := manifest.ExecuteYAML(ctx, yaml, images, cfg)
+	    if err != nil {
+		panic(err)
+	    }
+
+	manifest.OutputYAML(os.Stdout, files)
+        // Output:
+	// apiVersion: sources.knative.dev/v1
+	// kind: SinkBinding
+	// metadata:
+	//   name: foo
+	//   namespace: bar
+	// spec:
+	//   sink:
+	//     ref:
+	//          kind: sinkkind
+	// 			namespace: bar
+	// 			name: sinkname
+	// 			apiVersion: sinkversion
+	//     uri: uri/parts
```

Comment:
See the error message:

Result got:
```
got:
apiVersion: sources.knative.dev/v1
kind: SinkBinding
metadata:
  name: foo
  namespace: bar
spec:
  sink:
    ref:
      apiVersion: sinkversion
      kind: sinkkind
      namespace: bar
      name: sinkname
    uri: uri/parts
  subject:
    kind: <no value>
    apiVersion: <no value>
    namespace: bar
```
This is what your test expects:
```
want:
apiVersion: sources.knative.dev/v1
kind: SinkBinding
metadata:
  name: foo
  namespace: bar
spec:
  sink:
    ref:
         kind: sinkkind
			namespace: bar
			name: sinkname
			apiVersion: sinkversion
    uri: uri/parts
}
```
Values are correct, it is just that the indentation and field order is wrong, so we need to adapt the comment under `Output:` to the correct indentation and correct field order (basically just copy `got` from the error message as comment under `Output:`

Does this help?

Reply:
@taniaduggal